### ISSUE TITLE
🐞 Bug report: Flexbox Container & Margins causes horizontal scrollbar

### ISSUE THREAD

**Issue by hogash** [elementor/elementor#19614]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

When i add margins (LEFT in particular) to a Container, it creates a horizontal scrollbar.

Screenshot:

<img width="1918" alt="Markup 2022-08-25 at 22 26 14" src="https://user-images.githubusercontent.com/1725925/186751428-0c1bd1b6-54cf-400a-b25e-e604ee463efc.png">

### Steps to reproduce

Add the Container in the page and simply add LEFT margin.

### Isolating the problem

- [X] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently using the steps above.

### System Info

-

**Comment by hogash** [comment]
A potential solution is to **subtract the side margins**:

```
:is(.elementor-section-wrap, [data-elementor-id]) > .e-container {
  --margin-right: auto;
  --margin-left: auto;
  max-width: min(100%, calc( var(--width) - var(--margin-left, auto) - var(--margin-right, auto) ));
}
```

**Comment by T313C0mun1s7** [comment]
We have a site we just published and just noticed this bug. This seems to be the exact issue we are seeing. Using the Hello theme. I just subscribed to see how long it will take for somebody to get assigned to this issue.

**Comment by pwnoobz** [comment]
On my website, if I set the margin to anything or zero when the main container is full width, It goes to the left. It's basically the same problem but different values.

**Comment by TOTLD** [comment]
This appears to still be a current issue. I'm seeing this behaviour on a fresh Elementor Pro install with Hello theme. Level 2 support replied that they are aware of the bug, but have no ETA on a fix. No alternative was offered - has anyone found a workaround for this? @hogash your hack doesn't seem to work any more unfortunately.

**Comment by sheldonlkumm** [comment]
Same issue here. Fresh install, Elementor Pro and Hello theme. Adding left and right margin value seems to only add left margin (almost like it pushes the section outside of the screen on the right-hand side).

**Comment by TOTLD** [comment]
> 

@wookiesareppl2 please bother Elementor support about this, especially if you are a paying Pro customer like us. 

Tier 2 and 3 support have both simply told me there's no workaround and no ETA on a fix. One suggested that I post here to earn the bug "more traction" with the development team.

Since this has been an open bug report since August 2022 (and still nobody assigned to it) I'm not sure how we will achieve that besides frequent reminders.

**Comment by wookiesareppl2** [comment]
> > 
> 
> @wookiesareppl2 please bother Elementor support about this, especially if you are a paying Pro customer like us. 
> 
> Tier 2 and 3 support have both simply told me there's no workaround and no ETA on a fix. One suggested that I post here to earn the bug "more traction" with the development team.
> 
> Since this has been an open bug report since August 2022 (and still nobody assigned to it) I'm not sure how we will achieve that besides frequent reminders.

I tried to reach out to support yesterday but there were no available agents, so I will be trying again today.

Seeing as we purchased the expert plan subscription with 25 licences with the plan of using Elementor for all our company's projects moving forward, hopefully I manage to get a solution to this as I do feel something like adding margins being a basic functionality requirement does constitute something that should be solved as soon as possible, especially seeing as I noticed this while starting a new client project yesterday and had me run straight into a wall.

I did see this issue with the previous client site, but due to the design I was able to implement padding to the same intended result. In this case however I need a background color to the section but need to have margin around it. Padding would have the background color span the entire screen still and only move the contents within of course. So I would either need to remove the flexbox functionality (which I don't think I'm willing to do as the layout was designed with this in mind) or repurpose some designed sections to overcome this issue. With signed-off mockups this becomes an issue though.

I guess a part of me just feels like maybe I'm doing something wrong still, especially seeing as I would expect to see more people reporting this issue online as well?

Anyway, will post update once I have one 👍🏻

**Comment by TOTLD** [comment]
@wookiesareppl2 sounds like we have very similar situations! Signed off designs that expect to be able to use basic web layout variables (side margins) since that's part of every website under the sun.

FYI my ugly workaround for now has been:

- add an extra wrapping container around what *should* be the main parent container
- set the new parent container to have its content boxed and centered horizontally (similar effect to side-margins)
- set the child container to 100% width
- use width breakpoints (create more if needed) to fudge your design with pixel and/or VW max-widths of the parent container

There are a number of reasons this is a bad way to get to a similar result. But after being burnt using custom CSS hacks on Elementor containers before that end up redundant, at least this method uses all stock Elementor builder to achieve.

**Comment by wookiesareppl2** [comment]
Hey @TOTLD, so I received feedback from Elementor support:

*******************************************************************
Hi,
Thank you for contacting us.

My apologies for missing your chat.

We are working on fixing this known bug in the following updates.

As a workaround, you can change that container to full width, then click on the advanced tab and, unlink the chain, use the mouse to drag the right side to the left, it will not change the left side margin, so if you want them both, add the chain back and drag either side.

Please review the following screencast -
- https://www.loom.com/share/ebb9c8a353114a84bbab4dbdc0b0d704?sid=6b8ec4ea-5353-4697-a772-3fc76e30559f

My apologies for the inconvenience.
Please let me know if this works for you
*******************************************************************

So basically you give the container a full-width layout. If a margin is needed you give the container a 90% width for example. Note the container will always align left and cannot be centered. You then have to add a left margin to the container to center the contents. A VERY ugly workaround, but I have tested it as working should this help anyone else until they manage to release an update on this.

**Comment by leadlamp1** [comment]
I still have this problem - has there been a fix AND is it only on the Hello theme? I will not be using this theme going forward if it is the reason.. Anyone have a better minimal starter theme to go forward with?

**Comment by MassStash** [comment]
I've been running into this issue for some time now, and just working around it. I finally was tired of it, and started looking for a fix.  So just adding to the list of confused people, of why this isn't fixed yet haha. Margin on top container seems like a common enough thing, so wild the issue is maybe setup dependent? You'd think everyone on elementor would have this issue, or at least on the flexbox containers? interesting.

**Comment by benravetta** [comment]
Can't believe this is still not fixed. If you want to use margin on top container to create a border, whole website shifts to the right and you end up with off screen scrolling and a broken layout.

**Comment by wookiesareppl2** [comment]
> I still have this problem - has there been a fix AND is it only on the Hello theme? I will not be using this theme going forward if it is the reason.. Anyone have a better minimal starter theme to go forward with?

I have not tested in any other theme other than the Hello theme, so feel free to let us know if using another theme does not produce this issue as it might be useful to others stumbling onto this thread :)

**Comment by wookiesareppl2** [comment]
> I still have this problem - has there been a fix AND is it only on the Hello theme? I will not be using this theme going forward if it is the reason.. Anyone have a better minimal starter theme to go forward with?

I have not tested in any other theme other than the Hello theme, so feel free to let us know if using another theme does not produce this issue as it might be useful to others stumbling on this thread :)

**Comment by wookiesareppl2** [comment]
> I still have this problem - has there been a fix AND is it only on the Hello theme? I will not be using this theme going forward if it is the reason.. Anyone have a better minimal starter theme to go forward with?

I have not tested in any other theme other than the Hello theme, so feel free to let us know if using another theme does not produce this issue as it might be usefull to others stumbling on this thread :)

> I've been running into this issue for some time now, and just working around it. I finally was tired of it, and started looking for a fix. So just adding to the list of confused people, of why this isn't fixed yet haha. Margin on top container seems like a common enough thing, so wild the issue is maybe setup dependent? You'd think everyone on elementor would have this issue, or at least on the flexbox containers? interesting.

Yes, you would think! I have not needed to use this since I posted the issue, so not sure if this is fixed yet, but since we have a recent comment by @benravetta it indicates that it is still an issue...

**Comment by UmerBusy** [comment]
I still have this problem how could I resolve this issue? Anyone know please help

**Comment by LogoLogics2** [comment]
Wow, this issue still excists, october 2024, how can they not fix this?
I already waited a long time, before starting to use flexbox containers, now I finally do, and things like this just do not work!
I thought it would be safe now, since the beta fase is long gone....

**Comment by MassStash** [comment]
> Wow, this issue still excists, october 2024, how can they not fix this? I already waited a long time, before starting to use flexbox containers, now I finally do, and things like this just do not work! I thought it would be safe now, since the beta fase is long gone....

Ping from your reply, I too am STILL shocked this hasn't been figured out lol. However, I have seen this in 2 other editors one not wordpress at all, since running into this in element and following this thread I think years ago now. So that's interesting to note. Must be a bug with the internet eh? Or should I say not a hug, feature? 🤦‍♂️ 😂

**Comment by LogoLogics2** [comment]
@MassStash,

Yes, shocked is the word, haha.
Especially because they said in june 2023, they would come up with a fix:

**"We are working on fixing this known bug in the following updates"**

I will take my chances and send in a support ticket, will let you know the answer over here.

**Comment by MassStash** [comment]
> @MassStash,
> 
> Yes, shocked is the word, haha. Especially because they said in june 2023, they would come up with a fix:
> 
> **"We are working on fixing this known bug in the following updates"**
> 
> I will take my chances and send in a support ticket, will let you know the answer over here.

lol yeeaaaa, bit strange because It's still annoying when I go to add margin to top level container and it does the jacked to right margin thing... Then remember it's adjust sub container instead. 

Oh nice, that would be much appreciated for the effort. I wish you / us all luck! haha

**Comment by LogoLogics2** [comment]
@MassStash,

So, you work around it by adding subcontainers and adjust them instead?
Could you elaborate, please?


**For me, I used the workaround:**

_As a workaround, you can change that container to full width, then click on the advanced tab and, unlink the chain, use the mouse to drag the right side to the left, it will not change the left side margin, so if you want them both, add the chain back and drag either side._

But so far, that does solve the mobile version, but messes up the desktop and laptop versions, because setting the container to full width in Mobile mode, changes it on all other screens as well.

PS. had first response from EM support, asking me to send them screenshots or recordings, so I won't get bored today, haha.

**Comment by LogoLogics2** [comment]
Hi,

I got quick help from Elementor support, and for me, it is solved!
They told me I should not use margins to containers or headings, to push them to the right, but paddings.

Margins will push the element outside the container, creating that scrollbar/shift on mobile.
Padding will push to the right or left, but keep the element always in the container, without creating that scrollbar.
